libpod/container.go

```diff
@@ -536,12 +536,63 @@ func (c *Container) Unmount() error {
 
 // Pause pauses a container
 func (c *Container) Pause() error {
-	return ErrNotImplemented
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	if err := c.syncContainer(); err != nil {
+		return err
+	}
+
+	if c.state.State == ContainerStatePaused {
+		return errors.Wrapf(ErrCtrStateInvalid, "%q is already paused", c.ID())
+	}
+	if c.state.State == ContainerStateRunning {
```

Comment:
As below, can you add a single if to return an error if state is bad?

Reply:
Also, I think we can also pause Created and Stopped containers (they might have exec processes running)

Reply:
fixed